flow/scheduler.go

```diff
@@ -496,10 +498,28 @@ func (ff *flowFunction) checkInputRing() (n uint32) {
 	return n
 }
 
-func (ff *flowFunction) checkOutputRing() (n uint32) {
+func (ff *flowFunction) checkOutputRing(max uint32) bool {
```

Comment:
This function should be renamed to isOutputRingClonable, the same for input ring